File: app/src/main/java/com/example/theduckcardchatsystem/ui/groups/AddContactsFragment.java
package com.example.theduckcardchatsystem.ui.groups;

import android.content.Intent;
import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.os.Bundle;
import android.util.Base64;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.Observer;
import androidx.lifecycle.ViewModelProvider;
import androidx.lifecycle.ViewModelProviders;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.example.theduckcardchatsystem.R;
import com.example.theduckcardchatsystem.ui.model.CommonModel;
import com.example.theduckcardchatsystem.ui.model.MembersId;
import com.example.theduckcardchatsystem.ui.room.viewmodel.MembersViewModel;
import com.example.theduckcardchatsystem.viewmodel.AddContactsViewModel;
import com.example.theduckcardchatsystem.viewmodel.SettingsViewModel;
import com.google.android.material.floatingactionbutton.FloatingActionButton;
import com.google.firebase.database.DataSnapshot;

import java.io.ByteArrayInputStream;
import java.io.InputStream;
import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;
import de.hdodenhof.circleimageview.CircleImageView;

public class AddContactsFragment extends Fragment {
    private AddContactsViewModel addContactsViewModel;
    @BindView(R.id.add_contacts_recycle_view)
    RecyclerView addContactsRecycler;
    @BindView(R.id.profile_photo_create_group)
    CircleImageView profile_photo_image;

    private List<CommonModel> commonModelList = new ArrayList<>();
    public static List<CommonModel> contactsList = new ArrayList<>();
    private MembersViewModel membersViewModel;
    private SettingsViewModel settingsViewModel;

    public View onCreateView(@NonNull LayoutInflater inflater,
                             ViewGroup container, Bundle savedInstanceState) {
        View root = inflater.inflate(R.layout.fragment_add_contacts, container, false);
        ButterKnife.bind(this,root);
        addContactsViewModel = new ViewModelProvider(this,ViewModelProvider
                .AndroidViewModelFactory.getInstance(getActivity().getApplication())).get(AddContactsViewModel.class);
        addContactsViewModel.initRecyclerView(addContactsRecycler,getActivity());
        membersViewModel = new ViewModelProvider(getActivity()
                ,ViewModelProvider.AndroidViewModelFactory.getInstance(getActivity().getApplication())).get(MembersViewModel.class);
        contactsList.clear();
        settingsViewModel = new ViewModelProvider(this,ViewModelProvider
                .AndroidViewModelFactory.getInstance(getActivity().getApplication())).get(SettingsViewModel.class);
        settingsViewModel.setSettingsRepository();
        settingsViewModel.dataSnapshotLiveData().observe(getActivity(), new Observer<DataSnapshot>() {
            @Override
            public void onChanged(DataSnapshot dataSnapshot) {
                if (dataSnapshot != null){
                    String photo = dataSnapshot.child("photourl").getValue(String.class);
                    if (photo == null){
                        profile_photo_image.setImageResource(R.drawable.ic_profile);
                    }else {
                        String imageDataBytes = photo.substring(photo.indexOf(",")+1);
                        InputStream stream = new ByteArrayInputStream(Base64.decode(imageDataBytes.getBytes(), Base64.DEFAULT));
                        Bitmap bitmap = BitmapFactory.decodeStream(stream);
                        profile_photo_image.setImageBitmap(bitmap);
                    }
                }
            }
        });

        return root;
    }
    @OnClick(R.id.add_contacts_btn_next)
    void btnNext(){
        if(contactsList.isEmpty()){
            Toast.makeText(getActivity(), "Add a participant", Toast.LENGTH_SHORT).show();
        }else {
            for (CommonModel it : contactsList){

                MembersId membersId = new MembersId(it.getUid());
                membersViewModel.insert(membersId);
                MembersId membersId1 = new MembersId(it.getUid());
                membersId.setId(membersId1.getId());
                membersViewModel.Update(membersId1);
            }

            Intent i  = new Intent(getActivity(),CreateGroupActivity.class);
            i.putExtra("size",contactsList.size());
            startActivity(i);


        }
    }

    @Override
    public void onStart() {
        super.onStart();
        membersViewModel.DeleteAll();
    }

}

File: app/src/main/java/com/example/theduckcardchatsystem/ui/groups/AddContactsFragment.kt
package com.example.theduckcardchatsystem.ui.groups

import com.example.theduckcardchatsystem.viewmodel.AddContactsViewModel
import butterknife.BindView
import com.example.theduckcardchatsystem.R
import androidx.recyclerview.widget.RecyclerView
import com.example.theduckcardchatsystem.ui.model.CommonModel
import com.example.theduckcardchatsystem.ui.room.viewmodel.MembersViewModel
import android.view.LayoutInflater
import android.view.ViewGroup
import android.os.Bundle
import butterknife.ButterKnife
import androidx.lifecycle.ViewModelProvider
import com.example.theduckcardchatsystem.ui.groups.AddContactsFragment
import android.graphics.Bitmap
import android.graphics.BitmapFactory
import butterknife.OnClick
import android.widget.Toast
import com.example.theduckcardchatsystem.ui.model.MembersId
import android.content.Intent
import android.util.Base64
import android.view.View
import androidx.fragment.app.Fragment
import com.example.theduckcardchatsystem.ui.groups.CreateGroupActivity
import com.example.theduckcardchatsystem.viewmodel.SettingsViewModel
import de.hdodenhof.circleimageview.CircleImageView
import java.io.ByteArrayInputStream
import java.io.InputStream
import java.util.ArrayList

class AddContactsFragment : Fragment() {
    private var addContactsViewModel: AddContactsViewModel? = null

    @JvmField
    @BindView(R.id.add_contacts_recycle_view)
    var addContactsRecycler: RecyclerView? = null

    @JvmField
    @BindView(R.id.profile_photo_create_group)
    var profile_photo_image: CircleImageView? = null
    private val commonModelList: List<CommonModel> = ArrayList()
    private var membersViewModel: MembersViewModel? = null
    private var settingsViewModel: SettingsViewModel? = null
    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?, savedInstanceState: Bundle?
    ): View? {
        val root = inflater.inflate(R.layout.fragment_add_contacts, container, false)
        ButterKnife.bind(this, root)
        addContactsViewModel = ViewModelProvider(
            this, ViewModelProvider.AndroidViewModelFactory.getInstance(
                activity!!.application
            )
        ).get(
            AddContactsViewModel::class.java
        )
        addContactsViewModel!!.initRecyclerView(addContactsRecycler, activity)
        membersViewModel = ViewModelProvider(
            activity!!,
            ViewModelProvider.AndroidViewModelFactory.getInstance(activity!!.application)
        ).get(
            MembersViewModel::class.java
        )
        contactsList.clear()
        settingsViewModel = ViewModelProvider(
            this, ViewModelProvider.AndroidViewModelFactory.getInstance(
                activity!!.application
            )
        ).get(
            SettingsViewModel::class.java
        )
        settingsViewModel!!.setSettingsRepository()
        settingsViewModel!!.dataSnapshotLiveData().observe(activity!!) { dataSnapshot ->
            if (dataSnapshot != null) {
                val photo = dataSnapshot.child("photourl").getValue(
                    String::class.java
                )
                if (photo == null) {
                    profile_photo_image!!.setImageResource(R.drawable.ic_profile)
                } else {
                    val imageDataBytes = photo.substring(photo.indexOf(",") + 1)
                    val stream: InputStream = ByteArrayInputStream(
                        Base64.decode(
                            imageDataBytes.toByteArray(),
                            Base64.DEFAULT
                        )
                    )
                    val bitmap = BitmapFactory.decodeStream(stream)
                    profile_photo_image!!.setImageBitmap(bitmap)
                }
            }
        }
        return root
    }

    @OnClick(R.id.add_contacts_btn_next)
    fun btnNext() {
        if (contactsList.isEmpty()) {
            Toast.makeText(activity, "Add a participant", Toast.LENGTH_SHORT).show()
        } else {
            for (it in contactsList) {
                val membersId = MembersId(it.uid)
                membersViewModel!!.insert(membersId)
                val membersId1 = MembersId(it.uid)
                membersId.id = membersId1.id
                membersViewModel!!.Update(membersId1)
            }
            val i = Intent(activity, CreateGroupActivity::class.java)
            i.putExtra("size", contactsList.size)
            startActivity(i)
        }
    }

    override fun onStart() {
        super.onStart()
        membersViewModel!!.DeleteAll()
    }

    companion object {
        @JvmField
        var contactsList: MutableList<CommonModel> = ArrayList()
    }
}